src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetCoreMvc2Integration.cs

```diff
@@ -168,7 +170,7 @@ public AspNetCoreMvc2Integration(object actionDescriptor, object httpContext)
                 instrumentedMethod =
                     MethodBuilder<Action<object, object, object, object>>
                        .Start(moduleVersionPtr, mdToken, opCode, nameof(BeforeAction))
-                       .WithConcreteTypeName(DiagnosticSource)
+                       .WithConcreteType(DiagnosticSourceType)
```

Comment:
Leaving a comment for posterity: Perhaps we could rename this to `WithDeclaringType` to indicate the type that holds the method definition, since this case is static and "ConcreteType" makes me immediately think of an object instance.

Reply:
Agreed. I was gonna suggest a quick pass at refactoring `MethodBuilder` in light of the new methods we've added recently. I'd be happy to do that if y'all agree.

Reply:
In a separate PR, though.

Reply:
In a separate PR, though.